entrypoint.sh

```diff
@@ -41,43 +57,55 @@ chown nginx:nginx /var/cache/nginx
 mkdir -p /var/tmp/nginx
 chown nginx:nginx /var/tmp/nginx
 
+#create vhost directory
+mkdir -p /etc/nginx/vhosts/
+
+# Copy nginx main conf file.
+cp templates/nginx.conf /etc/nginx/nginx.conf
+
 # Process templates
-for t in /templates/*
+upstreamId=0
+for t in "${DOMAINSARRAY[@]}"
 do
-  dest="/etc/nginx/$(basename $t)"
-  echo "Rendering template of $dest"
-  sed -e "s/\${DOMAIN}/${DOMAIN}/g" \
-      -e "s/\${UPSTREAM}/${UPSTREAM}/" \
-      $t > $dest
+  dest="/etc/nginx/vhosts/$(basename "${t}").conf"
+  echo "Rendering template of $t in $dest"
+  sed -e "s/\${DOMAIN}/${t}/g" \
+      -e "s/\${UPSTREAM}/${UPSTREAMARRAY[upstreamId]}/" \
+      /templates/vhost.sample.conf > "$dest"
+  upstreamId=$((upstreamId+1))
 done
 
+
+for d in "${DOMAINSARRAY[@]}"
+do
+
 # Initial certificate request, but skip if cached
-if [ ! -f /etc/letsencrypt/live/${DOMAIN}/fullchain.pem ]; then
-  letsencrypt certonly \
-    --domain ${DOMAIN} \
-    --authenticator standalone \
-    ${SERVER} \
-    --email "${EMAIL}" --agree-tos
-fi
+  if [ ! -f /etc/letsencrypt/live/"${d}"/fullchain.pem ]; then
+    letsencrypt certonly \
+      --domain "${d}" \
+      --standalone \
+      --email "${EMAIL}" --agree-tos
+  fi
```

Comment:
Rather than running `letsencrypt certonly` for each domain, it's better to run `letsencrypt certonly` a single time and pass multiple `--domain` arguments. That way you'll generate a single TLS certificate with many subject alternate names and not use up your five-certificates-in-seven-days quota.